mz_bokeh_package/components/plot_settings.py

```diff
@@ -357,12 +362,20 @@ def _point_shape(self, value: str):
             create_glyph(**kwargs)
 
         # remove previous renderers
-        self._plot.renderers = self._plot.renderers[len(glyph_kwargs):]
-
-        # Update legend
+        excluded_renderers_list=[]
+        for renderer in self._plot.renderers:
+            # Deny excluded list renderers removal , such as trend line.
+            if renderer.name in self._excluded_renderers:
+                excluded_renderers_list.append(renderer)
+        self._plot.renderers = self._plot.renderers[(len(self._plot.renderers)-len(glyph_kwargs)):] + \
+                               excluded_renderers_list
+
+        # Update legend: scatter point renderers are not added to the legend
+        number_of_legend_items = len(self._plot.renderers) - len(self._plot.legend.items)
```

Comment:
This looks very strange to me. You first remove the renderers of the trend line from the list that `_get_glyph_kwargs` produces, and then add them again. Maybe I need to read it a bit more carefully, but so far, this doesn't make much sense to me...